src/_gettsim/aggregation_numpy.py

```diff
@@ -120,13 +121,33 @@ def grouped_cumsum(column, group_id):
     )
     fail_if_dtype_of_group_id_not_int(group_id, agg_func="sum")
     fail_if_dtype_not_numeric_or_boolean(column, agg_func="sum")
-    if column.dtype in ["bool"]:
+    if column.dtype == bool:
         column = column.astype(int)
     out = npg.aggregate(group_id, column, func="cumsum", fill_value=0)
 
     return out
 
 
+def sum_values_by_index(
+    column: numpy.ndarray,
```

Comment:
For consistency, we could always use the type `npt.NDArray` instead of `numpy.ndarray`.